Upgrade `esbuild`

### Details
Lightning is currently on esbuild `0.17.13`. The latest version of esbuild, as of this writing is `0.25.0`.
We need to upgrade to the latest version to get all the improvements and fixes that have been added.

At the moment, trying to upgrade to `> 0.20` results in the following error when you try to `--minify` the assets:

```bash
✘ [ERROR] Could not resolve "path"

    node_modules/@typescript/vfs/dist/vfs.esm.js:575:17:
      575 │   return require(String.fromCharCode(112, 97, 116, 104));
          ╵                  ^

  The package "path" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "--platform=node" to do that, which will remove this error.

1 error
** (Mix) `mix esbuild default --minify` exited with 1
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -21,7 +21,7 @@ and this project adheres to
 
 ### Fixed
 
-## [v2.11.0-pre.0] - 2025-03-12
+## [v2.11.0-pre.1] - 2025-03-12
 
 ### Added
 
@@ -36,6 +36,8 @@ and this project adheres to
   [#2941](https://github.com/OpenFn/lightning/issues/2941)
 - Use dropdown instead of modal for the log level filter
   [#2980](https://github.com/OpenFn/lightning/issues/2980)
+- Upgrade esbuild to 0.25.0
+  [#2962](https://github.com/OpenFn/lightning/issues/2962)
 
 ### Fixed
 

--- a/config/config.exs
+++ b/config/config.exs
@@ -99,6 +99,7 @@ config :esbuild,
          --loader:.woff2=file
          --loader:.ttf=copy
          --format=esm --splitting --bundle
+         --external:path
          --jsx=automatic
          --tsconfig=tsconfig.browser.json
          --target=es2020

